WarehouseManagement.Core/Services/DifferenceService.cs

```diff
@@ -0,0 +1,177 @@
+        {
+            ReceptionNumber = model.ReceptionNumber,
+            InternalNumber = model.InternalNumber,
+            ActiveNumber = model.ActiveNumber,
+            Comment = model.Comment,
+            Count = model.Count,
+            Status = model.Status,
+            TypeId = model.DifferenceTypeId,
+            ZoneId = model.ZoneId,
+            DeliveryId = model.DeliveryId,
+            CreatedByUserId = userId
+        };
+
+        await repository.AddAsync(difference);
+        await repository.SaveChangesAsync();
+    }
+
+    public async Task DeleteAsync(int id)
+    {
+        if (!await ExistsByIdAsync(id))
+        {
+            throw new KeyNotFoundException(DifferenceWithIdNotFound);
+        }
+
+        await repository.SoftDeleteById<Difference>(id);
+        await repository.SaveChangesAsync();
+    }
+
+    public async Task EditAsync(int id, DifferenceFormDto model, string userId)
+    {
+        if (!await ExistsByIdAsync(id))
+        {
+            throw new KeyNotFoundException(DifferenceWithIdNotFound);
+        }
+
+        var difference = await repository.GetByIdAsync<Difference>(id);
+        
+        difference!.ReceptionNumber = model.ReceptionNumber;
+        difference.InternalNumber = model.InternalNumber;
+        difference.ActiveNumber = model.ActiveNumber;
+        difference.Comment = model.Comment;
+        difference.Count = model.Count;
+        difference.Status = model.Status;
+        difference.TypeId = model.DifferenceTypeId;
+        difference.ZoneId = model.ZoneId;
+        difference.DeliveryId = model.DeliveryId;
+        difference.LastModifiedAt = DateTime.UtcNow;
+        difference.LastModifiedByUserId = userId;
+
+        await repository.SaveChangesWithLogAsync();
+    }
+
+    public async Task<bool> ExistsByIdAsync(int id)
+    {
+        return await repository.GetByIdAsync<Difference>(id) != null;
+    }
+
+    public async Task<IEnumerable<DifferenceDto>> GetAllAsync(PaginationParameters paginationParams)
+    {
+        Expression<Func<Difference, bool>> filter = v =>
+            EF.Functions.Like(v.ReceptionNumber, $"%{paginationParams.SearchQuery}%")
+            || EF.Functions.Like(v.InternalNumber, $"%{paginationParams.SearchQuery}%")
+            || EF.Functions.Like(v.ActiveNumber, $"%{paginationParams.SearchQuery}%");
+
+        var models = await repository
+            .AllReadOnly<Difference>()
+            .Paginate(paginationParams, filter)
+            .Select(d => new DifferenceDto()
+            {
+                Id = d.Id,
+                ReceptionNumber = d.ReceptionNumber,
+                InternalNumber = d.InternalNumber,
+                ActiveNumber = d.ActiveNumber,
+                Comment = d.Comment,
+                Count = d.Count,
+                Status = d.Status,
+                Type = d.Type.Name,
+                Zone = d.Zone.Name,
+                DeliverySystemNumber = d.Delivery.SystemNumber
```

Comment:
what about AdminComment?

Reply:
You sure you want to be included in the form? Wouldn't it be better to be passed on Finish Processing/No Differences?